groups/bsl/bslstl/bslstl_variant.h

```diff
@@ -0,0 +1,3338 @@
+        // allocator to supply memory to the created (value) object are
+        // disallowed by the compiler.  Note that if the constructor of 'TYPE'
+        // throws an exception this object is left in a
+        // 'valueless_by_exception' state.
+#else   //  BSL_VARIANT_FULL_IMPLEMENTATION
+
+    template <class TYPE>
+    TYPE& emplace();
+    template <class TYPE, class ARG>
+    TYPE& emplace(BSLS_COMPILERFEATURES_FORWARD_REF(ARG) arg);
+        // Create an object of alternative 'TYPE' created in place using the
+        // specified 'args' and return a reference providing modifiable access
+        // to the underlying 'TYPE' object.  If this 'variant' object already
+        // contains an object ('false == valueless_by_exception()'), that
+        // object is destroyed before the new object is created.  The allocator
+        // specified at the construction of this 'variant' object is used to
+        // supply memory to the value object.  Attempts to explicitly specify
+        // via 'args' another allocator to supply memory to the created (value)
+        // object are disallowed by the compiler.  Note that if the constructor
+        // of 'TYPE' throws an exception this object is left in a
+        // 'valueless_by_exception' state.
+
+    template <size_t INDEX>
+    typename variant_alternative<INDEX, variant<HEAD, TAIL...> >::type&
+    emplace();
+    template <size_t INDEX, class ARG>
+    typename variant_alternative<INDEX, variant<HEAD, TAIL...> >::type&
+    emplace(BSLS_COMPILERFEATURES_FORWARD_REF(ARG) arg);
+        // Create an object of alternative with index 'INDEX' created in place
+        // using the specified 'args' and return a reference providing
+        // modifiable access to the underlying object.  If this 'variant'
+        // object already contains an object
+        // ('false == valueless_by_exception()'), that object is destroyed
+        // before the new object is created.  The allocator specified at the
+        // construction of this 'variant' object is used to supply memory to
+        // the value object.  Attempts to explicitly specify via 'args' another
+        // allocator to supply memory to the created (value) object are
+        // disallowed by the compiler.  Note that if the constructor of 'TYPE'
+        // throws an exception this object is left in a
+        // 'valueless_by_exception' state.
+#endif  //BSL_VARIANT_FULL_IMPLEMENTATION
+    // 20.7.3.3, assignment
+    variant& operator=(const variant& rhs);
+        // If 'rhs' holds the same alternative type as this object, assign to
+        // this alternative object the value of the alternative contained in
+        // the specified 'rhs' object. Otherwise, destroy the currently held
+        // alternative object, if any, and construct an object of alternative
+        // type held by 'rhs', if any, using the value of 'rhs' alternative
+        // object and the allocator held by this object. The allocators of this
```

Comment:
modified to 
```
        // If 'rhs' holds the same alternative type as this object, assign to
        // this alternative object the value of the alternative contained in
        // the specified 'rhs' object. Otherwise, destroy the currently held
        // alternative object, if any, and construct an object of alternative
        // type held by 'rhs', if any, using the value of 'rhs' alternative
        // object and the allocator held by this object.  If the construction
        // of the alternative throws an exception, this object is left in a
        // 'valueless_by_exception' state.  The allocators of this object and
        // 'rhs' both remain unchanged.  The contents of 'rhs' are either
        // copy-constructed into or copy-assigned to this object.  'rhs' object
        // remains unchanged.
   ```
   
   Same wording applied to other `operator=`. This may need some massaging when I fix the exception safety. We should return to this when those changes are fixed - leaving open until such time.

Reply:
you will be implementing all the noexcept-specifiers as part of this PR?

Reply:
no, I was a bit vague there. I was referring to the variant "construct a temporary variant if copy can throw" trickery when assigning from another variant object.